src/tabs/chat/ChatTabState.tsx

```diff
@@ -0,0 +1,17 @@
+import { OnlinePlayerMetadata } from "common/io/OnlinePlayerMetadata";
+
+export interface Message {
+  type: "CONSOLE" | "CHAT" | "ERROR" | "NOTIFICATION" | "CLIENT";
+  message: string;
+}
+
+export interface ChatTabState {
+  messages?: Message[];
+  commandToSend?: string;
+  commands?: string[];
+  messageSendStatus?: "SENDING" | "SENT" | "ERROR" | "NONE";
+  errorMessage?: string;
+  selectedRecipient?: string;
+  commandMode?: boolean;
+  onlinePlayers?: OnlinePlayerMetadata[];
```

Comment:
Remember that these are like global variables. Try to minimize the amount of variables in the state if you can.